test/api.js

```diff
@@ -0,0 +1,279 @@
+		serial: true
+	});
+
+	api.run()
+		.then(function () {
+			t.ok(api.serial);
+			t.same(api.passCount, 2);
+		});
+});
+
+test('circular references on assertions do not break process.send', function (t) {
+	t.plan(2);
+
+	var api = new Api([path.join(__dirname, 'fixture/circular-reference-on-assertion.js')]);
+
+	api.run()
+		.then(function () {
+			t.same(api.failCount, 1);
+			t.true(/'c'.*?'d'/.test(api.errors[0].error.message));
+		});
+});
+
+test('change process.cwd() to a test\'s directory', function (t) {
+	t.plan(1);
+
+	var api = new Api([path.join(__dirname, 'fixture/process-cwd.js')]);
+
+	api.run()
+		.then(function () {
+			t.same(api.passCount, 1);
+		});
+});
+
+test('babel require hook only applies to the test file', function (t) {
+	t.plan(3);
+
+	var api = new Api([path.join(__dirname, 'fixture/babel-hook.js')]);
+
+	api.on('error', function (data) {
+		t.same(data.name, 'SyntaxError');
+		t.true(/Unexpected token/.test(data.message));
+	});
+
+	api.run()
+		.catch(function (err) {
+			t.ok(err);
+		});
+});
+
+test('unhandled promises will throw an error', function (t) {
+	t.plan(3);
+
+	var api = new Api([path.join(__dirname, 'fixture/loud-rejection.js')]);
+
+	api.on('error', function (data) {
+		t.same(data.name, 'Error');
+		t.true(/You can\'t handle this!/.test(data.message));
+	});
+
+	api.run()
+		.then(function () {
+			t.same(api.passCount, 1);
+		});
+});
+
+test('uncaught exception will throw an error', function (t) {
+	t.plan(3);
+
+	var api = new Api([path.join(__dirname, 'fixture/uncaught-exception.js')]);
+
+	api.on('error', function (data) {
+		t.same(data.name, 'Error');
+		t.true(/Can\'t catch me!/.test(data.message));
+	});
+
+	api.run()
+		.then(function () {
+			t.same(api.passCount, 1);
+		});
+});
+
+test('stack traces for exceptions are corrected using a source map', function (t) {
+	t.plan(4);
+
+	var api = new Api([path.join(__dirname, 'fixture/source-map-file.js')]);
+
+	api.on('error', function (data) {
+		t.true(/Thrown by source-map-fixtures/.test(data.message));
+		t.match(data.stack, /^.*?at.*?run\b.*source-map-fixtures.src.throws.js:1.*$/m);
+		t.match(data.stack, /^.*?at\b.*source-map-file.js:11.*$/m);
+	});
+
+	api.run()
+		.then(function () {
+			t.same(api.passCount, 1);
+		});
+});
+
+test('stack traces for exceptions are corrected using a source map, taking an initial source map for the test file into account', function (t) {
+	t.plan(4);
+
+	var api = new Api([path.join(__dirname, 'fixture/source-map-initial.js')]);
+
+	api.on('error', function (data) {
+		t.true(/Thrown by source-map-fixtures/.test(data.message));
+		t.match(data.stack, /^.*?at.*?run\b.*source-map-fixtures.src.throws.js:1.*$/m);
+		t.match(data.stack, /^.*?at\b.*source-map-initial-input.js:7.*$/m);
+	});
+
+	api.run()
+		.then(function () {
+			t.same(api.passCount, 1);
+		});
+});
+
+test('stack traces for exceptions are corrected using a source map, found via a pragma', function (t) {
```

Comment:
This test was removed in https://github.com/sindresorhus/ava/commit/386416846bdb7b8aae84e1b0d8cf77da0bb31fa4